multines.js

```diff
@@ -49,57 +51,38 @@ function register (server, options, next) {
         deliverMap[path] = buildDeliver(socket, topic)
       }
 
-      wrapSubscribe(socket, path, params, (err) => {
-        if (err) {
-          return next(err)
-        }
-
-        mq.on(topic, deliverMap[path], next)
-      })
+      await wrapSubscribe(socket, path, params)
+      await util.promisify(mq.on.bind(mq))(topic, deliverMap[path])
     }
 
-    options.onUnsubscribe = (socket, path, params, next) => {
-      wrapUnsubscribe(socket, path, params, (err) => {
-        if (err) {
-          return next(err)
-        }
+    options.onUnsubscribe = async (socket, path, params) => {
+      await wrapUnsubscribe(socket, path, params)
 
-        const deliverMap = socket[kDeliver] || {}
-        socket[kDeliver] = deliverMap
+      const deliverMap = socket[kDeliver] || {}
+      socket[kDeliver] = deliverMap
 
-        if (!deliverMap[path]) {
-          next()
-          return
-        }
+      if (!deliverMap[path]) {
+        return
+      }
 
-        mq.removeListener(path.replace(/^\//, ''), deliverMap[path], function () {
-          setImmediate(next)
-        })
-      })
+      await util.promisify(mq.removeListener.bind(mq))(path.replace(/^\//, ''), deliverMap[path])
```

Comment:
can we just store `util.promisify(mq.removeListener.bind(mq))` in a variable at the top?